src/com/carolinarollergirls/scoreboard/core/impl/ScoreBoardImpl.java

```diff
@@ -83,13 +83,21 @@ protected void valueChanged(PermanentProperty prop, Object value, Object last, F
                 lc.setMaximumTime(getRulesets().getLong(Rule.LINEUP_DURATION));
             }
         } else if (prop == Value.UPCOMING_JAM) {
-            removeAll(Period.NChild.JAM);
             if (value instanceof Jam) {
                 add(Period.NChild.JAM, (Jam)value);
             }
         }
     }
 
+    @Override
+    protected void itemRemoved(AddRemoveProperty prop, ValueWithId item) {
+      if (prop == NChild.PERIOD) {
+        // If the current period was removed, assign
+        // the upcoming jam to the now current period.
+        getCurrentPeriod().getCurrentJam().setNext(getUpcomingJam());
```

Comment:
This should not be needed. Next-links should be updated under the hood if a jam is removed. If they aren't, that's a bug elsewhere.

Reply:
This is a whole period being removed, and the jam->jam link crosses period boundaries so needs special handling.

Reply:
That case should be covered by the under the hood code.

Reply:
Where should that be happening exactly?